### DISCUSSION TITLE
C# Wrapper for minecraft-data

### DISCUSSION THREAD

**Discussion by codemonkey85** [psu-de/MineSharp#30] [Ideas]
Hello,

I was just wondering if you'd consider contributing MineSharp.Data as a wrapper for [minecraft-data](https://github.com/PrismarineJS/minecraft-data).

For my part, it would be useful to have that as a NuGet package so I could use the data in a Blazor app I'm working on.

**Comment by psu-de** [comment]
Hey,
I released the first nuget packages two days ago, including [MineSharp.Data](https://www.nuget.org/packages/MineSharp.Data).
For minecraft-data: MineSharp is listed under project's using minecraft-data, because it is not really a wrapper for it.

**Reply by codemonkey85** [reply]
Ah, cool! Doesn't seem to work super well in Blazor WebAssembly, unfortunately.

```csharp
protected override void OnInitialized()
{
    TestMineSharp();
}

private void TestMineSharp()
{
    var data = MinecraftData.FromVersion("1.20.2");
    for (var i = 0; i < data.Blocks.TotalBlockStateCount; i++)
    {
        Blocks.Add(data.Blocks[i]);
    }

    Blocks = [.. Blocks.OrderBy(b => b.DisplayName)];
}
```

<img width="1110" alt="image" src="https://github.com/psu-de/MineSharp/assets/1414036/00cfaf60-f83f-4167-b08b-672a9d1f4170">

**Reply by psu-de** [reply]
Hm MineSharpData loads a lot of static data when created, i guess WebAssembly can't handle that. I used to have a wrapper that loads the json files from minecraft-data directly, not using static data at all. It's pretty messy, but if you'd want to give it a try, I'll look for the code.

**Reply by codemonkey85** [reply]
As it happens, [I'm working on my own](https://github.com/codemonkey85/minecraft-data-csharp)!
